cirq/sim/simulator.py

```diff
@@ -177,6 +177,74 @@ def compute_samples_displays_sweep(
         return compute_displays_results
 
 
+class SimulatesAmplitudes(metaclass=abc.ABCMeta):
+    """Simulator that computes final amplitudes of given bitstrings.
+
+    Given a circuit and a list of bitstrings, computes the amplitudes
+    of the given bitstrings in the state obtained by applying the circuits
+    to the all zeros state. Implementors of this interface should implement
+    the compute_amplitudes_sweep method.
+    """
+
+    def compute_amplitudes(
+            self,
+            program: Union[circuits.Circuit, schedules.Schedule],
+            bitstrings: np.ndarray,
+            param_resolver: 'study.ParamResolverOrSimilarType' = None,
+            qubit_order: ops.QubitOrderOrList = ops.QubitOrder.DEFAULT,
+    ) -> List[complex]:
+        """Computes the desired amplitudes.
+
+        The initial state is assumed to be the all zeros state.
+
+        Args:
+            program: The circuit or schedule to simulate.
+            bitstrings: The bitstrings whose amplitudes are desired, input
```

Comment:
@cduck can you comment on how this would play with your qudit work?

Reply:
This is fine as it can easily be changed from bools to ints.

I think this API could be improved by making it match more closely with how other simulators return measurement results.  E.g. outputs=[[mesurement key 'a'=0b100110, key 'b'=5, ...], ...]

Is it possible for this kind of simulator to not measure some qubits at the end?

Reply:
Usually with this kind of simulator qubits are not measured at all.

Reply:
ok.